concourse/pipelines/linux-image-build.jsonnet

```diff
@@ -145,49 +160,61 @@ local ELImgBuildJob(image, workflow) = imgbuildjob {
 local RHUAImgBuildJob(image, workflow) = imgbuildjob {
   image: image,
   workflow: workflow,
+  daily: false,
 
   // Append var to Daisy image build task
-  buildtask+: { vars+: [
-    'instance_service_account=rhui-builder@rhel-infra.google.com.iam.gserviceaccount.com',
-  ] },
+  buildtask: RHUIImgBuildTask(workflow, '((.:gcs-url))'),
 };
 
 local CDSImgBuildJob(image, workflow) = imgbuildjob {
+  local acme_server = 'dv.acme-v02.api.pki.goog',
+  local acme_email = 'bigcluster-guest-team@google.com',
+  local rhui_project = 'google.com:rhel-infra',
+
   image: image,
   workflow: workflow,
+  daily: false,
   extra_tasks: [
     {
-      task: 'get-rhui-tls-key',
-      config: gcp_secret_manager.get_secret { secret_name: 'rhui-tls-key' },
-    },
-    // These subsequent invocations need to declare gcp-secret-manager as output and as an input, so we will
-    // layer all the secrets into the same output. This dynamic setting of inputs can't be done with the base
-    // YAML task config.
-    {
-      task: 'get-acme-hmac',
-      config: gcp_secret_manager.get_secret {
-        secret_name: 'rhui-acme-hmac',
-        inputs+: gcp_secret_manager.get_secret.outputs,
+      task: 'get-acme-account-json',
+      config: gcp_secret_manager.getsecrettask {
+        secret_name: 'rhui-acme-account-json',
+        project: rhui_project,
+        output_path: 'accounts/' + acme_server + '/' + acme_email + '/account.json',
       },
     },
     {
-      task: 'get-acme-key-id',
-      config: gcp_secret_manager.get_secret {
-        secret_name: 'rhui-acme-kid',
-        inputs+: gcp_secret_manager.get_secret.outputs,
+      task: 'get-rhui-tls-key',
+      config: gcp_secret_manager.getsecrettask {
+        secret_name: 'rhui-tls-key',
+        project: rhui_project,
+        output_path: 'accounts/' + acme_server + '/' + acme_email + '/keys/' + acme_email + '.key',
+
+        // Layer onto the same output as previous task
+        inputs+: gcp_secret_manager.getsecrettask.outputs,
       },
     },
     {
-      task: 'certbot-provision-tls-crt',
-      file: 'guest-test-infra/concourse/tasks/certbot-rhui.yaml',
+      task: 'lego-provision-tls-crt',
+      config: lego.legotask {
+        domains: ['rhui.googlecloud.com', 'staging-rhui.googlecloud.com'],
+        acme_server: acme_server,
+        email: acme_email,
+        project: rhui_project,
+        input: 'gcp-secret-manager',
+      },
     },
   ],
 
   // Append var to Daisy build task
-  buildtask+: { vars+: [
-    'instance_service_account=rhui-builder@rhel-infra.google.com.iam.gserviceaccount.com',
-    'tls_cert_path=../../../../tls-output/rhui.crt',
-  ] },
+  buildtask: RHUIImgBuildTask(workflow, '((.:gcs-url))') {
```

Comment:
Not necessarily for this PR, but I noticed extra_tasks follows snake case and this is squished into one non-delimited string. Should we make the args consistent?

Reply:
sorry, i don't quite follow what you mean